src/codec.ts

```diff
@@ -27,22 +27,33 @@ export class NoiseHandshakeMessage extends MessageV0 implements Message {
  * Used in the pairing object for encoding the messages exchanged
  * during the handshake process
  */
-export class NoiseHandshakeEncoder implements Encoder {
+export class NoiseHandshakeEncoder implements IEncoder {
   /**
    * @param contentTopic content topic on which the encoded WakuMessages will be sent
    * @param hsStepResult the result of a step executed while performing the handshake process
+   * @param ephemeral makes messages ephemeral in the Waku network
    */
-  constructor(public contentTopic: string, private hsStepResult: HandshakeStepResult) {}
+  constructor(public contentTopic: string, private hsStepResult: HandshakeStepResult, public ephemeral: boolean = true) {}
```

Comment:
As I understand nature of the exchange I think `ephemeral` should be set to `true`. 
@fryorcraken , what do you think?

Reply:
I am not totally sure. I think there was discussions of being able to resume a noise handshake off store but not sure it would work out due to possible timeouts on each side.

@s1fr0 @richard-ramos, thoughts?

Reply:
Thanks @s1fr0 . At this stage I think it's fine to assume both online and set as ephemeral. We can then review when one express the need to allow async pairing for a specific use case.

Reply:
Saved it for future consideration here https://github.com/waku-org/js-noise/issues/20